src/components/Layout/FormPageLayout.tsx

```diff
@@ -141,16 +169,69 @@ function FormSections<F>(props: Pick<FormPageLayoutProps<F>, "formSections" | "f
   );
 }
 
-function LeftNav<F>(props: Pick<FormPageLayoutProps<F>, "formSections">) {
+function LeftNav<F>(props: { sectionsWithRefs: SectionWithRefs<F>[] }) {
+  const { sectionsWithRefs } = props;
   const tids = useTestIds(props);
 
+  // Ignore sections that don't have titles defined
+  const sectionWithTitles = useMemo(
+    () => sectionsWithRefs.filter(({ section }) => !!section.title),
+    [sectionsWithRefs],
+  );
+
+  const activeSection = useActiveSection(sectionWithTitles);
+
   return (
     <aside css={Css.gr(2).gc("2 / 3").sticky.topPx(headerHeightPx).px3.df.fdc.gap1.$} {...tids}>
-      {/* Content TODO here: [SC-67747] to implement new NavLink component and section scroll-to behavior */}
+      {sectionWithTitles.map((sectionWithRef) => (
+        <SectionNavLink
+          key={`nav-${sectionWithRef.sectionKey}`}
+          sectionWithRef={sectionWithRef}
+          activeSection={activeSection}
+        />
+      ))}
     </aside>
   );
 }
 
+// Use inset box shadow rather than thick border to avoid the button text reflowing when the border is applied
+const activeStyles = Css.baseBd.boxShadow(`inset 3px 0px 0 0px ${Palette.Blue600}`).$;
+const hoverStyles = Css.bgBlue50.baseBd.blue900.boxShadow(`inset 3px 0px 0 0px ${Palette.Blue900}`).$;
+const defaultFocusRingStyles = Css.relative.z2.bshFocus.$;
+
+function SectionNavLink<F>(props: { sectionWithRef: SectionWithRefs<F>; activeSection: string | null }) {
```

Comment:
I originally started implementing this as a more generic `LeftNavLink` component, but it didn't really make sense on it's own since we don't have any examples of this type of link style outside this "scroll to section" concept 🤷 
Maybe we pull the internals out of this at a later point if we do?